graphql_compiler/cost_estimation/analysis.py

```diff
@@ -56,6 +63,21 @@ def _convert_int_interval_to_field_value_interval(
     return Interval(lower_bound, upper_bound)
 
 
+def _get_location_vertex_path(location: BaseLocation) -> VertexPath:
+    """Get the VertexPath for a BaseLocation pointing at a vertex."""
+    if location.field is not None:
+        raise AssertionError(f"Location {location} represents a field.")
+
+    if isinstance(location, Location):
+        return location.query_path
+    elif isinstance(location, FoldScopeLocation):
+        return location.base_location.query_path + tuple(
+            "{}_{}".format(direction, name) for direction, name in location.fold_path
+        )
+    else:
+        raise AssertionError("Unexpected location encountered: {}".format(location))
```

Comment:
I'd probably pull this out of the `else`.

The issue is that if one of the `if-elif` branches is taken but doesn't end in a return (not today, not tomorrow, but sometime in the future), we might fall through to the end of the function and get an implicit `return None`. With the last statement of the function being `raise AssertionError()` that will be impossible.

The other alternative is to construct a result object in the `if-elif` branches, and then return that object as the last line of the function. That might look even cleaner.